src/parsing_routines_matrixnetworks.jl

```diff
@@ -122,3 +122,125 @@ function loadrxnetwork(ft::MatrixNetwork,
+                U <: SparseMatrixCSC{Int,Int} ,V <: AbstractVector,W <: AbstractVector}
+    rateexprs::S
+    cxstoichmat::T
+    in_mat::U  # all elements are integer always in incidence matrix
+    species::V
+    params::W
+end
+ComplexSparseMatrixNetwork(rateexprs,cxstoichmat,in_mat;species=Any[],params=Any[]) =
+           ComplexSparseMatrixNetwork(rateexprs,cxstoichmat,in_mat,species,params)
+
+
+# for Dense matrices version
+function loadrxnetwork(cmn::T) where {T <: ComplexMatrixNetwork}
+    numspecs, numcomp = size(cmn.cxstoichmat)
+    @assert all(>=(0),cmn.cxstoichmat)
+    @assert numcomp == size(cmn.in_mat, 1)
+    @assert all(∈([-1,0,1]),cmn.in_mat)
+    numrxs = size(cmn.in_mat, 2)
+
+    ModelingToolkit.@parameters t
+
+    isempty(cmn.species) ? species = [funcsym(:S,t,i) for i = 1:numspecs] : species = cmn.species
+
+    rn = Vector{Reaction}(undef,numrxs)
+    sc_ind = argmin(cmn.in_mat, dims=1)  # cartesian indices of substrate complexes
+    pc_ind = argmax(cmn.in_mat, dims=1)  # cartesian indicies of products complexes
+
+    for i ∈ 1:numrxs
+
+        # substrate index for i'th reaction in species(rn)
+        ss_ind = findall(!iszero, @view cmn.cxstoichmat[:,sc_ind[i][1]])
+
+        # products index for i'th reaction in species(rn)
+        ps_ind = findall(!iszero, @view cmn.cxstoichmat[:,pc_ind[i][1]])
+
+        if isempty(ss_ind) && !isempty(ps_ind)
+            rn[i] = Reaction(cmn.rateexprs[i], nothing, species[ps_ind],
+                        nothing, cmn.cxstoichmat[ps_ind,pc_ind[i][1]])
+
+        elseif !isempty(ss_ind) &&  isempty(ps_ind)
+            rn[i] = Reaction(cmn.rateexprs[i], species[ss_ind], nothing,
+                        cmn.cxstoichmat[ss_ind,sc_ind[i][1]], nothing)
+        else
+            rn[i] = Reaction(cmn.rateexprs[i], species[ss_ind],species[ps_ind],
+                        cmn.cxstoichmat[ss_ind,sc_ind[i][1]],
+                        cmn.cxstoichmat[ps_ind,pc_ind[i][1]])
+        end
+    end
+    @named rs = ReactionSystem(rn,t,species, cmn.params)
+    return ParsedReactionNetwork(rs,nothing)
+end
+
+# for sparse matrices version
+function loadrxnetwork(cmn::T) where {T <: ComplexSparseMatrixNetwork}
+    numspecs, numcomp = size(cmn.cxstoichmat)
+    @assert all(>=(0),cmn.cxstoichmat)
+    @assert numcomp == size(cmn.in_mat, 1)
+    @assert all(∈([-1,0,1]),cmn.in_mat)
+    numrxs = size(cmn.in_mat, 2)
+
+    cp = cmn.cxstoichmat;   inmat= cmn.in_mat
+
+    ModelingToolkit.@parameters t
+
+    isempty(cmn.species) ? species = [funcsym(:S,t,i) for i = 1:numspecs] : species = cmn.species
+
+    rn = Vector{Reaction}(undef,numrxs)
+    sc_ind = findall(x -> x == -1, inmat)  # cartesian indices of substrate complexes
+    pc_ind = findall(x -> x == 1, inmat)  # cartesian indicies of products complexes
+
+
+    for i ∈ 1:numrxs
+        # substrate index for i'th reaction in species(rn)
+        ss_ind = rowvals(cp)[cp.colptr[sc_ind[i][1]]:cp.colptr[sc_ind[i][1]+1]-1]
```

Comment:
Can't you just use `nzrange` to get the values? (Look at the docs for `nzrange` to see how to access values in a given column of a sparse matrix.) You shouldn't be directly accessing the underlying field `colptr` but using the functions Julia provides as an interface to access the entries (`rowvals`, `nonzeros` and `nzrange`).

Reply:
done.. using `nxrange and rowvals` and `nonzeros` for good learning experience with sparsearrays